static/app/components/alert.tsx

```diff
@@ -104,92 +172,67 @@ const alertStyles = ({
   `;
 };
 
-const StyledTextBlock = styled('span')`
-  line-height: ${p => p.theme.text.lineHeightBody};
-  position: relative;
-  flex: 1;
+const Wrap = styled('div')<AlertProps & {hovered: boolean}>`
+  ${alertStyles}
 `;
 
-const MessageContainer = styled('div')`
+const IconWrapper = styled('div')`
   display: flex;
+  height: calc(${p => p.theme.fontSizeMedium} * ${p => p.theme.text.lineHeightBody});
+  padding-top: ${space(1.5)};
+  padding-bottom: ${space(1.5)};
+  padding-left: ${space(2)};
+  padding-right: ${space(1)};
+  box-sizing: content-box;
+  align-items: center;
+`;
+
+const ContentWrapper = styled('div')`
   width: 100%;
 `;
 
-const ExpandContainer = styled('div')`
+const MessageContainer = styled('div')<{
+  showIcon: boolean;
+  showTrailingItems: boolean;
+}>`
+  display: flex;
+  width: 100%;
+  padding-top: ${space(1.5)};
+  padding-bottom: ${space(1.5)};
+  padding-left: ${p => (p.showIcon ? '0' : space(2))};
+  padding-right: ${p => (p.showTrailingItems ? space(1.5) : space(2))};
+`;
+
+const Message = styled('span')`
+  line-height: ${p => p.theme.text.lineHeightBody};
+  position: relative;
+  flex: 1;
+`;
+
+const TrailingItems = styled('div')`
+  height: calc(${p => p.theme.fontSizeMedium} * ${p => p.theme.text.lineHeightBody});
   display: grid;
-  padding-top: ${space(1)};
+  grid-auto-flow: column;
+  grid-template-rows: 100%;
+  align-items: center;
+  gap: ${space(1)};
 `;
 
-const ExpandIcon = styled(props => (
-  <IconWrapper {...props}>{<IconChevron />}</IconWrapper>
-))`
-  transform: ${props => (props.isExpanded ? 'rotate(0deg)' : 'rotate(180deg)')};
-  justify-self: flex-end;
+const TrailingItemsWrap = styled(TrailingItems)`
```

Comment:
any way to get rid of the wrap 🤔

Reply:
created two wraps because I wanted to stop event propagation inside `<TrailingItems />` (the area containing the two buttons below), but not on `IconChevron`, which is in `<TrailingItemsWrap />` but not in `<TrailingItems />`.

<img width="713" alt="Screen Shot 2022-04-11 at 10 33 03 AM" src="https://user-images.githubusercontent.com/44172267/162796991-089f04d4-ab8d-422d-a36d-d5098dbe7192.png">

so one way to make this simpler is to apply the grid styles to `<TrailingItemsWrap />`, and then just add `display: contents` to `<TrailingItems />`. This way, everything inside `<TrailingItems />` will be grid items. Problem with this is that there's only partial support for `display: contents` in Safari. This is what it looks like in Safari:

<img width="648" alt="Screen Shot 2022-04-11 at 10 38 20 AM" src="https://user-images.githubusercontent.com/44172267/162797759-13f97944-8f4b-413d-b23d-ebd8de7fb23f.png">